meteor_packages/mats-common/imports/startup/server/data_match_util.js

```diff
@@ -350,6 +404,20 @@ const getMatchedDataSet = function (dataset, curveInfoParams, appParams, binStat
                     }
                 }
             }
+        } else if (isScalar && (curveDiffs[curveIndex] === undefined || curveDiffs[curveIndex] === null)) {
+            // need to recalculate the primary statistic with the newly matched hits, false alarms, etc.
+            dataLength = data[independentVarName].length;
+            for (di = 0; di < dataLength; di++) {
+                if (data.subSquareDiffSum[di] instanceof Array) {
+                    const squareDiffSum = matsDataUtils.sum(data.subSquareDiffSum[di]);
+                    const NSum = matsDataUtils.sum(data.subNSum[di]);
+                    const obsModelDiffSum = matsDataUtils.sum(data.subObsModelDiffSum[di]);
+                    const modelSum = matsDataUtils.sum(data.subModelSum[di]);
+                    const obsSum = matsDataUtils.sum(data.subObsSum[di]);
+                    const absSum = matsDataUtils.sum(data.subAbsSum[di]);
```

Comment:
Given my other comment, I appreciate the `const` here!

Reply:
I do use `const` for variables that don't change! There are some holdouts from early MATS but I'm gradually finding and changing them. As stated earlier, I can't do that for those temp arrays because they change with each loop iteration, and have things pushed to them, so they are not `const`s.